logstash-core/lib/logstash/api/lib/app/commands/system/plugins_command.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+require "app/command"
+
+class LogStash::Api::PluginsCommand < LogStash::Api::Command
+
+  def run
+    { :total => plugins.count, :plugins => plugins }
+  end
+
+  private
+
+  def plugins
+    @plugins ||= find_plugins_gem_specs.map do |spec|
+      { :name => spec.name, :version => spec.version.to_s }
```

Comment:
see the issue for ref specs on the implementation.

On Wed, Mar 16, 2016 at 10:53 AM Pere Urbon-Bayes <pere.urbon@elastic.co>
wrote:

> we only spoke about adding now the name and the version. we can definitely
> add more info in next iterations.
>
> On Wed, Mar 16, 2016 at 10:52 AM João Duarte <notifications@github.com>
> wrote:
>
>> In
>> logstash-core/lib/logstash/api/lib/app/commands/system/plugins_command.rb
>> <https://github.com/elastic/logstash/pull/4794#discussion_r56306233>:
>>
>> > @@ -0,0 +1,26 @@
>> > +# encoding: utf-8
>> > +require "app/command"
>> > +
>> > +class LogStash::Api::PluginsCommand < LogStash::Api::Command
>> > +
>> > +  def run
>> > +    { :total => plugins.count, :plugins => plugins }
>> > +  end
>> > +
>> > +  private
>> > +
>> > +  def plugins
>> > +    @plugins ||= find_plugins_gem_specs.map do |spec|
>> > +      { :name => spec.name, :version => spec.version.to_s }
>>
>> no dependencies of the plugin? :(
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/elastic/logstash/pull/4794/files/13d4bf00ff037b36907d0946411f324f0c218180#r56306233>
>>
>